src/Frame.cc

```diff
@@ -998,192 +1008,227 @@ bool Frame::FillSpatialProfileData(int region_id, CARTA::SpatialProfileData& spa
     auto tile_size = [](int tile_index, int total_size) { return std::min(TILE_SIZE, total_size - tile_index); };
 
     int x, y;
-    _cursor.ToIndex(x, y); // convert float to index into image array
+    _point_regions[region_id].ToIndex(x, y); // convert float to index into image array
     float cursor_value(0.0);
 
-    if (_image_cache_valid) {
-        bool write_lock(false);
-        tbb::queuing_rw_mutex::scoped_lock cache_lock(_cache_mutex, write_lock);
-        cursor_value = _image_cache[(y * _width) + x];
-        cache_lock.release();
-    } else if (_loader->UseTileCache()) {
-        int tile_x = tile_index(x);
-        int tile_y = tile_index(y);
-        auto tile = _tile_cache.Get(TileCache::Key(tile_x, tile_y), _loader, _image_mutex);
-        auto tile_width = tile_size(tile_x, _width);
-        cursor_value = (*tile)[((y - tile_y) * tile_width) + (x - tile_x)];
-    }
-
-    // set message fields
-    spatial_data.set_x(x);
-    spatial_data.set_y(y);
-    spatial_data.set_channel(CurrentZ());
-    spatial_data.set_stokes(CurrentStokes());
-    spatial_data.set_value(cursor_value);
-
-    // add profiles
-    std::vector<float> profile;
-    bool write_lock(false);
+    // Get point region spatial configs with respect to the stokes (key)
+    std::unordered_map<int, std::vector<CARTA::SetSpatialRequirements_SpatialConfig>> point_regions_spatial_configs;
+
+    for (auto& config : _point_regions_spatial_configs[region_id]) {
+        // Get stokes
+        std::string coordinate(config.coordinate());
+        int stokes;
+        if (!GetStokesTypeIndex(coordinate, stokes)) {
+            continue;
+        }
+        if (stokes < 0) {
```

Comment:
I would prefer to check for equality to `CURRENT_STOKES` explicitly. This should probably also be changed in `GetStokesTypeIndex`.